bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/ShellyBindingConstants.java

```diff
@@ -0,0 +1,255 @@
+    public static final String CHANNEL_METER_LASTMIN3 = "lastPower3";
+    public static final String CHANNEL_METER_TOTALKWH = "totalKWH";
+    public static final String CHANNEL_EMETER_TOTALRET = "returnedKWH";
+    public static final String CHANNEL_EMETER_REACTWATTS = "reactWatts";
+    public static final String CHANNEL_EMETER_VOLTAGE = "voltage";
+    public static final String CHANNEL_METER_TIMESTAMP = "timestamp";
+
+    public static final String CHANNEL_GROUP_LED_CONTROL = "led";
+    public static final String CHANNEL_LED_STATUS_DISABLE = "statusLed";
+    public static final String CHANNEL_LED_POWER_DISABLE = "powerLed";
+
+    public static final String CHANNEL_GROUP_SENSOR = "sensors";
+    public static final String CHANNEL_SENSOR_TEMP = "temperature";
+    public static final String CHANNEL_SENSOR_HUM = "humidity";
+    public static final String CHANNEL_SENSOR_LUX = "lux";
+    public static final String CHANNEL_SENSOR_FLOOD = "flood";
+    public static final String CHANNEL_SENSOR_MOTION = "motion";
+    public static final String CHANNEL_SENSOR_CHARGER = "charger";
+    public static final String CHANNEL_SENSOR_LASTUPDATE = "last_update";
+
+    public static final String CHANNEL_GROUP_SENSE_CONTROL = "control";
+    public static final String CHANNEL_SENSE_KEY = "key";
+
+    public static final String CHANNEL_GROUP_BATTERY = "battery";
+    public static final String CHANNEL_SENSOR_BAT_LEVEL = "batteryLevel";
+    public static final String CHANNEL_SENSOR_BAT_LOW = "lowBattery";
+    public static final String CHANNEL_SENSOR_BAT_VOLT = "voltage";
+
+    public static final String CHANNEL_GROUP_LIGHT_CONTROL = "control";
+    public static final String CHANNEL_LIGHT_COLOR_MODE = "mode";
+    public static final String CHANNEL_LIGHT_POWER = "power";
+    public static final String CHANNEL_LIGHT_DEFSTATE = "defaultState";
+    public static final String CHANNEL_GROUP_LIGHT_CHANNEL = "channel";
+
+    // Bulb/RGBW2 in color mode
+    public static final String CHANNEL_GROUP_COLOR_CONTROL = "color";
+    public static final String CHANNEL_COLOR_PICKER = "hsb";
+    public static final String CHANNEL_COLOR_FULL = "full";
+    public static final String CHANNEL_COLOR_RED = "red";
+    public static final String CHANNEL_COLOR_GREEN = "green";
+    public static final String CHANNEL_COLOR_BLUE = "blue";
+    public static final String CHANNEL_COLOR_WHITE = "white";
+    public static final String CHANNEL_COLOR_GAIN = "gain";
+    public static final String CHANNEL_COLOR_EFFECT = "effect";
+
+    // Bulb/RGBW2 in White Mode
+    public static final String CHANNEL_GROUP_WHITE_CONTROL = "white";
+    public static final String CHANNEL_COLOR_TEMP = "temperature";
+
+    // General
+    public static final String CHANNEL_EVENT_TRIGGER = "event";
+    public static final String CHANNEL_GROUP_SEPARATOR = "#";
+
+    public static final String OPENHAB_HTTP_PORT = "OPENHAB_HTTP_PORT";
```

Comment:
I'm not sure if this is the best way to get the port (The code in this binding uses `System.getenv`). The servlet itself can also provide the port number. Maybe @cweitkamp or @wborn can give an answer?

Reply:
You should use [HttpServiceUtil](https://github.com/openhab/openhab-core/blob/12b3c1030f2b8e320ebc839d2f94e48ffb0ffb0b/bundles/org.openhab.core/src/main/java/org/eclipse/smarthome/core/net/HttpServiceUtil.java#L38) to get the port (s) used by the HTTP service.

Reply:
hmm, I don't have the bundleContext at this point, class instance gets created by the thing

Reply:
solved